dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -70,6 +71,132 @@ def test_add_profilers(self):
+        data2 = pd.Series(["YO YO YO", "HELLO"])
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged',
+                'int': 1.0,
+                'float': 1.0,
+                'string': 'unchanged'
+            },
+            'data_type': ['int', 'text']
+        }
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+        
+        # Test same data types
+        data1 = pd.Series(['-2', '15', '1', '2'])
+        data2 = pd.Series(['5', '-1'])
+
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged',
+                'int': 'unchanged',
+                'float': 'unchanged',
+                'string': 'unchanged'
+            },
+             'data_type': 'unchanged',
+             'statistics': {
+                     'min': -1.0,
+                     'max': 10.0,
+                     'sum': 12.0,
+                     'mean': 2.0,
+                     'variance': 38.666666666666664,
+                     'stddev': 3.285085839971525
+                 }
+        }
+
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+        
+        # Test different compilers
+        data1 = pd.Series(['-2', '-1', '1', '2'])
+        data2 = pd.Series(['5', '15'])
+
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnStatsProfileCompiler(data2)
+        # Assert type error is properly called
+        with self.assertRaises(TypeError) as exc:
+            compiler1.diff(compiler2)
+        self.assertEqual(str(exc.exception),
+                         "`ColumnPrimitiveTypeProfileCompiler` and "
+                         "`ColumnStatsProfileCompiler` are not of the same "
+                         "profile compiler type.")
+        
+    def test_disabling_columns_during_primitive_diff(self):
+        data1 = pd.Series(['-2', '-1', '1', '2'])
+        data2 = pd.Series(['5', '15'])
+        options = StructuredOptions()
+
+        # Test disabled column in one compiler
+        options.int.is_enabled = False
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1,
+                                                                         options)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged', 
+                'float': 'unchanged', 
+                'string': 'unchanged', 
+                'int': [None, 1.0]}, 
```

Comment:
should we keep other results consistent with this format? 0 for not detected, 1 for detected, and None for non-existing

Reply:
Thats not what this is showing. This is showing that one profile had None for int ratio and the other profile had 1.0 for int ratio. That change could be made in the utils file in a diff PR tho.

Reply:
my concern is about the consistency of the results. Above we have `'int': 1` or `'int': -1`. Here we have `'int': [None, 1]`. Should we make a common output format?

Reply:
@JGSweets @ChrisWallace2020 : thoughts?. Also for my comment below.

Reply:
I think this is fine because if we just made it 1, that would imply `None` is synonymous with `0`, and I think that distinction should be kept intact